### ISSUE TITLE
[va-checkbox-group] - Add documentation for using header styles

### ISSUE THREAD

**Issue by danbrady** [department-of-veterans-affairs/vets-design-system-documentation#2204]
## Description
Add documentation and guidance for va-checkbox-group on design.va.gov for using the different header level styles. 

## Details

The [Checkbox Group Label Header](https://design.va.gov/storybook/?path=/docs/uswds-va-checkbox-group--default#label-header) story shows an instance of the label using a header 3 styles. We need design guidance on the [Checkbox page](https://design.va.gov/components/form/checkbox) to document this.

## Tasks
- [ ] Write or update component documentation for design.va.gov
- [ ] Coordinate with accessibility specialist to create or update accessibility guidance
- [ ] Request documentation draft review
- [ ] Once documentation is approved, publish to design.va.gov

## Acceptance Criteria
- [ ] Component documentation is published on design.va.gov

**Comment by humancompanion-usds** [comment]
Agreed. We should differentiate this variation versus Form Pattern Single.
